utils/src/main/scala/za/co/absa/enceladus/utils/validation/SchemaPathValidator.scala

```diff
@@ -108,6 +108,20 @@ object SchemaPathValidator {
     }
   }
 
+  /**
+    * Validate schema path data type is algebraic
+    *
+    * @param schema      A Spark schema
+    * @param fieldPath   A path to a field (e.g. "data.employees.employee.age")
+    * @return A list of ValidationErrors objects, each containing a column name and the list of errors and warnings
+    */
+  def validateSchemaPathAlgebraic(schema: StructType, fieldPath: String): Seq[ValidationIssue] = {
+    validateSchemaPathType(schema, fieldPath) {
+      case _: NumericType |  _: BooleanType => Seq.empty[ValidationIssue]
+      case k => Seq(ValidationError(s"The datatype '${k.typeName}' of '$fieldPath' field is not a algebraic type"))
```

Comment:
The error message might be confusing. `Algebraic type` is a not very common notion. I would rephrase it to be something like `The datatype '${k.typeName}' of '$fieldPath' field is neither numeric nor boolean`